pkg/agent/clean/clean.go

```diff
@@ -0,0 +1,325 @@
+/*
+The cleanup is designed to get a cluster that was imported to Rancher disconnected
+and cleanup some objects
+
+Remove the cattle-system namespace which houses the agent used to talk
+to Rancher
+
+Remove Rancher labels, finalizers and annotations from all namespaces
+
+If the cluster was imported to Rancher 2.1 or later cleanup roles, rolebindings,
+clusterRoles and clusterRoleBindings
+*/
+
+package clean
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/rancher/rancher/pkg/controllers/user/nslabels"
+	"github.com/sirupsen/logrus"
+	apierror "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+var (
+	// List of namespace labels that will be removed
+	nsLabels = []string{
+		nslabels.ProjectIDFieldLabel,
+	}
+
+	// List of namespace annotations that will be removed
+	nsAnnotations = []string{
+		"cattle.io/status",
+		"field.cattle.io/creatorId",
+		"lifecycle.cattle.io/create.namespace-auth",
+		nslabels.ProjectIDFieldLabel,
+	}
+)
+
+func Cluster() error {
+	// The sleep allows Rancher server time to finish updating ownerReferences
+	logrus.Info("Starting sleep for 1 min....")
+	time.Sleep(time.Duration(1) * time.Minute)
```

Comment:
What if instead of just sleeping for a minute, we tried a command in a loop?
Like:
```
for i < 60; i++ {
  err := removeCattleNamespace()
  if IsNotAuthorized(err) {
    some useful log statement
    sleep 1 second
    continue
}
```

Reply:
as discussed, just an env var to opt-in to sleeping

Reply:
Added